primitives/consensus/common/src/block_validation.rs

```diff
@@ -47,15 +48,23 @@ pub enum Validation {
 /// Type which checks incoming block announcements.
 pub trait BlockAnnounceValidator<B: Block> {
 	/// Validate the announced header and its associated data.
-	fn validate(&mut self, header: &B::Header, data: &[u8]) -> Result<Validation, Box<dyn Error + Send>>;
+	fn validate(
+		&mut self,
+		header: &B::Header,
+		data: &[u8],
+	) -> Box<dyn Future<Output = Result<Validation, Box<dyn Error + Send>>> + Send + Unpin>;
 }
 
 /// Default implementation of `BlockAnnounceValidator`.
 #[derive(Debug)]
 pub struct DefaultBlockAnnounceValidator;
 
 impl<B: Block> BlockAnnounceValidator<B> for DefaultBlockAnnounceValidator {
-	fn validate(&mut self, _h: &B::Header, _d: &[u8]) -> Result<Validation, Box<dyn Error + Send>> {
-		Ok(Validation::Success { is_new_best: false })
+	fn validate(
+		&mut self,
+		_: &B::Header,
+		_: &[u8],
+	) -> Box<dyn Future<Output = Result<Validation, Box<dyn Error + Send>>> + Send + Unpin> {
+		Box::new(async { Ok(Validation::Success { is_new_best: false }) }.boxed())
```

Comment:
You're double-boxing here. `boxed()` returns a `Pin<Box<...>>`

```suggestion
		async { Ok(Validation::Success { is_new_best: false }) }.boxed()
```